Buildings/Fluid/Storage/Plant/ChillerAndTank.mo

```diff
@@ -0,0 +1,418 @@
+        origin={0,-110}),   iconTransformation(
+        extent={{10,-10},{-10,10}},
+        rotation=180,
+        origin={110,-20})));
+  Buildings.Fluid.Movers.SpeedControlled_y pum2(
+    redeclare package Medium = Medium2,
+    per(pressure(dp=dp_nominal*{2,1.2,0}, V_flow=(mChi_flow_nominal +
+            mTan_flow_nominal)/1.2*{0,1.2,2})),
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    allowFlowReversal=true,
+    addPowerToMedium=false,
+    y_start=0,
+    T_start=T_CHWR_nominal) "Secondary pump" annotation (Placement(transformation(
+        extent={{10,-10},{-10,10}},
+        rotation=0,
+        origin={80,-20})));
+
+  Buildings.Fluid.Storage.Plant.BaseClasses.ReversiblePumpValveControl conPum2Gro
+    if allowRemoteCharging "Control block for secondary pump-valve group"
+    annotation (Placement(transformation(extent={{80,80},{60,102}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput booOnOff
+    if allowRemoteCharging "Plant status: true = on; false = off"
+                                                   annotation (Placement(
+        transformation(extent={{120,70},{100,90}}),
+        iconTransformation(extent={{-140,-120},{-100,-80}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput booFloDir
+    if allowRemoteCharging
+    "Flow direction: true = normal; false = reverse" annotation (Placement(
+        transformation(extent={{120,90},{100,110}}),   iconTransformation(
+          extent={{-140,-40},{-100,0}})));
+  Modelica.Blocks.Interfaces.RealInput set_mTan_flow if allowRemoteCharging
+    "Tank mass flow rate setpoint" annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=-90,
+        origin={80,130}),   iconTransformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={-110,20})));
+
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage val1(
+    redeclare package Medium = Medium2,
+    use_inputFilter=true,
+    y_start=0,
+    l=1E-5,
+    dpValve_nominal=0.1*dp_nominal,
+    m_flow_nominal=m2_flow_nominal)
+    if allowRemoteCharging
+    "Valve in series to the pump (normal direction)"
+    annotation (Placement(transformation(extent={{30,-20},{10,0}})));
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage val2(
+    redeclare package Medium = Medium2,
+    use_inputFilter=true,
+    y_start=0,
+    l=1E-5,
+    dpValve_nominal=0.1*dp_nominal,
+    m_flow_nominal=mTan_flow_nominal) if allowRemoteCharging
+    "Valve in parallel to the secondary pump (reverse direction)"
+    annotation (Placement(transformation(extent={{10,-80},{30,-60}})));
+  Buildings.Controls.OBC.CDL.Continuous.Switch swiFloDirPum1
+    if allowRemoteCharging "Switches off pum1 when tank charged remotely"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}},
+        rotation=-90,
+        origin={-10,90})));
+  Modelica.Blocks.Sources.Constant conZero(k=0) if allowRemoteCharging
+    "Constant y = 0"
```

Comment:
```suggestion
  Modelica.Blocks.Sources.Constant zero(k=0) if allowRemoteCharging
    "Constant 0"
```

Reply:
2be027b renamed it and 2a61cba improved its description.